standard/structs.md

```diff
@@ -476,3 +498,19 @@ Static constructors for structs follow most of the same rules as for classes. Th
 Automatically implemented properties ([§15.7.4](classes.md#1574-automatically-implemented-properties)) use hidden backing fields, which are only accessible to the property accessors.
 
 > *Note*: This access restriction means that constructors in structs containing automatically implemented properties often need an explicit constructor initializer where they would not otherwise need one, to satisfy the requirement of all fields being definitely assigned before any function member is invoked or the constructor returns. *end note*
+
+### §ref-like-field-initialization Ref safe scopes
+
+A `ref struct` may contain *ref_like field*s. A *ref_like field* is a *reference_variable* (§ref-span-safety). An instance of a `ref struct` may not be copied outside the *ref_safe_scope* (§ref-span-safety-escape-scopes) of any of its ref-like fields. The *ref_safe_scope* of a ref struct is determined from its initializing expression.
+
+The default value of ref-like fields is the value of the `default` expression, a null reference. The *ref_safe_scope* of a null reference is *caller_scope*. A `ref struct` initialized to its default value can be copied to the *caller_scope*.
+
+An instance of a `ref struct` `S` requires an initializing ref expression, `ref e` for any ref like fields. The *ref_safe_scope* of `e` determines the *ref_safe_scope* for `S`.
+
+- If `e` is a local variable, the *ref_safe_scope* of `S` is *block*.
+- If `e` is a formal parameter, the *ref_safe_scope* of `S` is *calling_method*. This includes when `e` is the `this` parameter.
+- If `e` designates a reference to a field, `e1.F`, the *ref_safe_scope* of `S` is the same as the *ref_safe_scope* of `e1`.
+- If `e` designates a reference to a conditional, `ref c ? ref e1 : ref e2`, the *ref_safe_scope* of `S` is the narrowest scope among the *ref_safe_scopes* of `e1` and `e2`.
+- If `e` is the result of a method invocation, `e1.M(e2, ...)`, the *ref_safe_scope* of `S` is the narrowest of the *ref_safe_scope*s of all argument expressions, including the receiver.
+
+For an assignment `e1 = e2`, if the type of `e1` is a `ref struct` type, then the *ref_safe_scope* of `e1` is the *ref_safe_scope* of `e2`.
```

Comment:
Not sure how assignment matters here. Would like to hear more in the meeting.

Reply:
Using a modified version of another example:

```csharp
    ReadOnlySpan<char> span1;
    ReadOnlySpan<char> span3;
    if (x is not null)
    {
        ReadOnlySpan<char> span2 = x.AsSpan(); // span2 ref_safe_scope is *method*.
        span1 = span2;     // span1 ref_safe_scope is *method* (same as span2)
        span3 = span1; // span3 ref_safe_scope is method (same as span1)
    }
}
```

Reply:
Discussion in meeting: a *variable* `v` has a *ref_safe_scope*, and if `v` can be assigned variable references which have different *ref_safe_scopes*, it would be odd for `v`'s *ref_safe_scope* to change.

As per https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.2/span-safety.md#language-constraints, the assignment part should be a constraint on what *can* be assigned, but a determination of the *ref_safe_scope* of the LHS.

Reply:
Also noted from @gafter: this would be safe_to_escape not ref_safe_to_escape in the previous way of explaining things.

Action 2023-04-19: @BillWagner to consider - and ping others where appropriate.

Reply:
These should be language constraints, as per https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.2/span-safety.md#language-constraints, not inference rules.

Reply:
Update per meeting notes.